docs/docs/development/hardware-integration/physical-layouts.md

```diff
@@ -0,0 +1,351 @@
+To use such layouts, import them and assign their `transform` and (optionally) `kscan` properties.
+
+### Physical Layout with Keys Example
+
+Here is an example of a physical layout for a 2x2 macropad:
+
+```dts
+#include <physical_layouts.dtsi>
+
+/ {
+    macropad_physical_layout: macropad_physical_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Macro Pad";
+        transform = <&default_transform>;
+        kscan = <&kscan0>;
+        keys  //                     w   h    x    y     rot    rx    ry
+            = <&key_physical_attrs 100 100    0    0       0     0     0>
+            , <&key_physical_attrs 100 100  100    0       0     0     0>
+            , <&key_physical_attrs 100 100    0  100       0     0     0>
+            , <&key_physical_attrs 100 100  100  100       0     0     0>
+            ;
+    };
+};
+```
+
+### Predefined Layout Usage Example
+
+Here is an example of using the predefined physical layouts for a 60% keyboard:
+
+```dts
+
+#include <layouts/common/60percent/all1u.dtsi>
+#include <layouts/common/60percent/ansi.dtsi>
+#include <layouts/common/60percent/hhkb.dtsi>
+#include <layouts/common/60percent/iso.dtsi>
+
+
+\\ Assigning the suitable matrix transforms
+&layout_60_ansi {
+    transform = <&ansi_transform>;
+    kscan = <&ansi_kscan>;
+};
+
+&layout_60_iso {
+    transform = <&iso_transform>;
+    kscan = <&iso_kscan>;
+};
+
+&layout_60_all1u {
+    transform = <&all_1u_transform>;
+    kscan = <&all_1u_kscan>;
+};
+
+&layout_60_hhkb {
+    transform = <&hhkb_transform>;
+    kscan = <&hhkb_kscan>;
+};
+```
+
+## Multiple Physical Layouts
+
+If a keyboard has multiple possible layouts (ex. you can snap off an outer column), then you should define multiple matrix transformations and multiple physical layouts - one for each possible layout.
+If necessary, you can also define multiple kscan instances.
+
+```dts
+// Needed if and only if keys property is used
+#include <physical_layouts.dtsi>
+
+/ {
+    default_layout: default_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_transform>;
+        kscan = <&kscan0>;
+        keys = // List of key positions, optional
+    };
+
+    alt_layout: alt_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Alternate Layout";
+        transform = <&alt_transform>;
+        kscan = <&alt_kscan0>;
+        keys = // List of key positions, optional
+    };
+};
+```
+
+### Position Map
+
+A position map is used for precise mapping between layouts. It is used to allow [ZMK Studio](../../features/studio.md) to accurately retain keymaps when switching layouts. A keyboard's position map has a child node for every potential layout of the keyboard, with the corresponding entries in `positions` property will be used to determine the mapping between layouts. A position map looks something like this:
```

Comment:
Is this missing an example after the last sentence?

Reply:
Yes, woops. Fixed.